src/Client.php

```diff
@@ -542,21 +549,32 @@ protected function send(array $data, array $tags = [])
      */
     protected function sendMessages(array $messages)
     {
-        $socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
-        if (!$socket) {
-            if ($this->throwExceptions) {
-                throw new ConnectionException($this, '(' . $errno . ') ' . $errstr);
-            } else {
-                trigger_error(
-                    sprintf('StatsD server connection failed (udp://%s:%d)', $this->host, $this->port),
-                    E_USER_WARNING
-                );
-                return $this;
+        if($this->socket === null) {
+            $this->socket = @fsockopen('udp://' . $this->host, $this->port, $errno, $errstr, $this->timeout);
+            if (!$this->socket) {
+                if ($this->throwExceptions) {
+                    throw new ConnectionException($this, '(' . $errno . ') ' . $errstr);
+                } else {
+                    trigger_error(
+                        sprintf('StatsD server connection failed (udp://%s:%d)', $this->host, $this->port),
+                        E_USER_WARNING
+                    );
+                }
             }
         }
-        $this->message = implode("\n", $messages);
-        @fwrite($socket, $this->message);
-        fclose($socket);
+        
+        if($this->socket) {
+            $this->message = implode("\n", $messages);
+            @fwrite($socket, $this->message);
```

Comment:
1. for timeout, it looks like the method: `stream_set_timeout` would be simpler.
1. for trigger_error, this code is supported back to php5.5 i think, so need to include something. PHP7 i believe you can choose to through an exception or not on trigger error, so inclined to keep it in